Implement map change tracking via state event

Nsgr opened the pull request:
This implements tracking which maps were played during a match via adding a new `MapMatchState` event based on `GbxRemoteEvent.BeginMap`.

Since this event is supposed to implement the `IMatchState` interface, it also needs to specify the current match status with the event. However, since this information cannot be inferred from the event itself, it is currently set to `Unknown`, which could cause side effects in other programs that do not expect the match state to differ from the usual cycle at any point in time. See src/Modules/MatchTrackerModule/Services/MatchTracker.cs at line 131.

Nsgr committed:
Implement map change tracking via state event

snixtho commented on src/Modules/MatchTrackerModule/Services/MatchTracker.cs:
```diff
@@ -114,6 +114,37 @@ await events.RaiseAsync(MatchTrackerEvent.StateTracked,
+        state = new MapMatchState
+        {
+            Status =
+                MatchStatus.Unknown, // FIXME Either we need to assume the match is running when the server map is changed (bad) or we override the status to unknown (also bad).
```
A match will always be running when the tracker is enabled, so setting to running is the correct one.

Nsgr replied:
This is now changed.

Nsgr committed:
Change match status on map change

A map change now always reports the match as running - adresses https://github.com/EvoEsports/EvoSC-sharp/pull/358#discussion_r2249386863
